### ISSUE TITLE
[Sitemap] The actual default value of sitemapName is incorrect, and the configuration does not take effect

### ISSUE THREAD

**Issue by Red-Asuka** [harlan-zw/nuxt-seo#159]
### Describe the bug

The default value written in the [document](https://nuxtseo.com/sitemap/api/config#sitemapname) for `sitemapName` is `sitemap.xml`. 

Regardless of whether I have set `sitemapName` or not, when using the `nuxt generate` command to generate static files, the actual `sitemapName` is always `sitemap_index.xml`.

```ts
export default defineNuxtConfig({
  devtools: { enabled: true },
  modules: ['@nuxtjs/i18n', '@nuxtjs/sitemap'],
  i18n: {
    vueI18n: './i18n.config.ts', // if you are using custom path, default
  },
  site: {
    url: 'https://example.com',
  },
  sitemap: {
    sitemapName: 'sitemap.xml',
  },
});

```

<img width="558" alt="image" src="https://github.com/harlan-zw/nuxt-seo/assets/38158783/f6378d1d-96d2-4e1b-a2b4-580a2652122b">

### Reproduction

https://stackblitz.com/edit/github-y149ef?file=nuxt.config.ts

### System / Nuxt Info

```Shell
------------------------------
- Operating System: Darwin
- Node Version:     v20.3.1
- Nuxt Version:     3.9.1
- CLI Version:      3.10.0
- Nitro Version:    2.8.1
- Package Manager:  pnpm@8.14.0
- Builder:          -
- User Config:      devtools, app, runtimeConfig, modules, i18n, site, plugins, css, postcss, vite, typescript, nitro, build, hooks
- Runtime Modules:  @nuxtjs/tailwindcss@6.10.4, @vueuse/nuxt@10.7.1, @nuxtjs/i18n@8.0.0, @pinia/nuxt@0.5.1, @zadigetvoltaire/nuxt-gtm@0.0.13, @nuxt/image@1.0.0-rc.1, @nuxtjs/sitemap@5.0.1, @nuxt/content@2.10.0
- Build Modules:    -
------------------------------
```

**Comment by harlan-zw** [comment]
The issue is that when you have an i18n config it will automatically switch to multiple sitemaps, this mode doesn't support changing the sitemap name.

You can disable this behavior with the following:

```ts
export default defineNuxtConfig({
  autoI18n: false
})
```

As for GSC not registering the sitemap, I'd say that it should fix itself. You can try manually importing the child sitemaps to get around it. Otherwise please share the URL and I'll double-check it's rendering properly.

**Comment by Red-Asuka** [comment]
> The issue is that when you have an i18n config it will automatically switch to multiple sitemaps, this mode doesn't support changing the sitemap name.
> 
> You can disable this behavior with the following:
> 
> ```ts
> export default defineNuxtConfig({
>   autoI18n: false
> })
> ```
> 
> As for GSC not registering the sitemap, I'd say that it should fix itself. You can try manually importing the child sitemaps to get around it. Otherwise please share the URL and I'll double-check it's rendering properly.

Thank you for your patience in explaining. I've tried importing the child sitemaps, but they still can't be recognized. I've used an online Sitemap validation tool to check, and it shows no problem. Here is the Sitemap URL of my site: https://mqttx.app/sitemap_index.xml. If it's not too much trouble, could you help me check it? Thanks again.

<img width="966" alt="image" src="https://github.com/harlan-zw/nuxt-seo/assets/38158783/2b756721-e867-4946-a4e0-69bf1e8b7163">

**Comment by harlan-zw** [comment]
I looked into your site and it seems okay, only interesting thing I noted was you're serving it as a gzip and some extra headers coming from your server, but it shouldn't be anything causing issues as far as I know.

I would recommend using the [nuxt-simple-robots](https://github.com/harlan-zw/nuxt-simple-robots) module so that the sitemap is added to robots.txt.

As far as I know this is a bug in GSC, it only shows success after the sitemap has been processed. If it hasn't resolved in the next couple of days let me know and I'll dig deeper into it

I have 3 sites using the module that have success, I just submitted a new one and see it's showing an error. So will use that a as a sanity check that it resolves itself.
